treeherder/config/settings.py

```diff
@@ -604,3 +604,35 @@
     "url": env.str('ELASTICSEARCH_URL', default=""),
     "index_prefix": ""
 }
+
+OPTION_COLLECTION_HASH_MAP = {
```

Comment:
Seems like these could be held in Redis rather than hard-coded here?  Not sure if you already discussed this though.

Reply:
No, didn't discuss that :)

Reply:
Yeah these and similar should be dynamically calculated. 

Early in the Treeherder days we made quite a few decision decisions that were in the name of performance, but ultimately made the architecture more complicated - and it turned out afterwards that the performance would have been just fine had we not done so. As such we're making a concerted effort moving forwards not to [prematurely optimise](https://en.wikipedia.org/wiki/Program_optimization#When_to_optimize).

For this case and similar below, I'd first do it the ORM way, then later once it's deployed and on New Relic we can see about query optimisation or adding Redis caching to certain calls.

Reply:
@edmorley when you say "For this case and similar below" are you also referring to the get_tree function in common.py - re dynamically create the tuple of id's based on the name rather than hard-code them?

Reply:
Yeah - sorry for not being more specific :-)

Reply:
Ah - yeah for "trunk" it's ok to hardcode a list of repository names that maps to (either server-side, client-side or both; treeherder and perfherder do a bit of all three) - the main thing is to not hardcode database primary key IDs or hashes that may change over time.

Reply:
Gotcha :)

Reply:
@camd  and @edmorley refactoring the get_tree code to fetch id's dynamically leads me back to where I started... Since I have to use query.objects.all (to return a list of id's for trunk), that queryset is evaluated when its used in the subsequent actual queries where they're needed and it takes a full 6 seconds to return results. This is why I created this function (and hardcoded id's), otherwise dynamically searching by repo name in my queries results in super slow returns and timeouts. Any suggestions on how to approach this?

Reply:
Nm, figured out a work-around :D